app/services/data_requests/fetch_cloudwatch_logs.rb

```diff
@@ -42,35 +41,36 @@ def query_string
     end
 
     def start_queries
-      dates.map do |date|
-        warn "Downloading logs for #{date}"
-        query_start = (date - 12.hours).to_i
-        query_end = (date + 36.hours).to_i
-
-        cloudwatch_client.start_query(
-          log_group_name: 'prod_/srv/idp/shared/log/events.log',
-          start_time: query_start,
-          end_time: query_end,
-          query_string: query_string,
-        ).query_id
-      end
-    end
-
-    def wait_for_query_results(query_ids)
       results = Concurrent::Array.new
-      thread_pool = Concurrent::FixedThreadPool.new(4)
+      thread_pool = Concurrent::FixedThreadPool.new(5)
 
-      query_ids.each do |query_id|
+      dates.each do |date|
         thread_pool.post do
-          results.push(wait_for_query_result(query_id))
+          warn "Downloading logs for #{date}"
+          query_id = start_query(date)
+          result = wait_for_query_result(query_id)
+          results.push(result)
```

Comment:
no need for some of these intermediate variables right?

```suggestion
          results.push(wait_for_query_result(start_query(date)))
```